solidity/contracts/bank/Bank.sol

```diff
@@ -0,0 +1,291 @@
+        address recipient,
+        uint256 amount
+    ) external {
+        uint256 currentAllowance = allowance[spender][msg.sender];
+        if (currentAllowance != type(uint256).max) {
+            require(
+                currentAllowance >= amount,
+                "Transfer amount exceeds allowance"
+            );
+            _approveBalance(spender, msg.sender, currentAllowance - amount);
+        }
+        _transferBalance(spender, recipient, amount);
+    }
+
+    /// @notice EIP2612 approval made with secp256k1 signature.
+    ///         Users can authorize a transfer of their balance with a signature
+    ///         conforming EIP712 standard, rather than an on-chain transaction
+    ///         from their address. Anyone can submit this signature on the
+    ///         user's behalf by calling the permit function, paying gas fees,
+    ///         and possibly performing other actions in the same transaction.
+    /// @dev    The deadline argument can be set to `type(uint256).max to create
+    ///         permits that effectively never expire.  If the `amount` is set
+    ///         to `type(uint256).max` then `transferBalanceFrom` will not
+    ///         reduce an allowance.
+    function permit(
+        address owner,
+        address spender,
+        uint256 amount,
+        uint256 deadline,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external {
+        /* solhint-disable-next-line not-rely-on-time */
+        require(deadline >= block.timestamp, "Permission expired");
+
+        // Validate `s` and `v` values for a malleability concern described in EIP2.
+        // Only signatures with `s` value in the lower half of the secp256k1
+        // curve's order and `v` value of 27 or 28 are considered valid.
+        require(
+            uint256(s) <=
+                0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF5D576E7357A4501DDFE92F46681B20A0,
+            "Invalid signature 's' value"
+        );
+        require(v == 27 || v == 28, "Invalid signature 'v' value");
+
+        bytes32 digest =
+            keccak256(
+                abi.encodePacked(
+                    "\x19\x01",
+                    DOMAIN_SEPARATOR(),
+                    keccak256(
+                        abi.encode(
+                            PERMIT_TYPEHASH,
+                            owner,
+                            spender,
+                            amount,
+                            nonce[owner]++,
+                            deadline
+                        )
+                    )
+                )
+            );
+        address recoveredAddress = ecrecover(digest, v, r, s);
+        require(
+            recoveredAddress != address(0) && recoveredAddress == owner,
+            "Invalid signature"
+        );
+        _approveBalance(owner, spender, amount);
+    }
+
+    /// @notice Increases balances of the provided `recipients` by the provided
+    ///         `amounts`. Can only be called by the Bridge.
```

Comment:
Anyone can call this function :)

Reply:
No because `increaseBalance` called by this function is `onlyBridge` :) This is tested [here](https://github.com/keep-network/tbtc-v2/pull/84/files#diff-2f96581e8d80fd83eef0ed8eed414e6445fe9fb7d9f0a673657250cc4539ee25R986-R997). I was thinking about adding `onlyBridge` here as well to make it explicit but my greed for gas savings won.

Reply:
100% should be mentioned in `@dev` but I'm all for those sweet gas savings

Reply:
Clarified in https://github.com/keep-network/tbtc-v2/pull/84/commits/dbbb8d81038dbf781f2222c888f26d3a52d9ef0c and then further optimized in https://github.com/keep-network/tbtc-v2/pull/84/commits/734ba437ef6fe899f8728cd7fcdc787aefc61252.